pkg/services/baremetal/host/host.go

```diff
@@ -1097,6 +1111,14 @@ func (s *Service) actionPreProvisioning(ctx context.Context) actionResult {
 	}
 	sshClient := s.scope.SSHClientFactory.NewClient(in)
 
+	out := sshClient.GetHostName()
+	hostName := trimLineBreak(out.StdOut)
+	if hostName != rescue {
+		// This is unexpected. We should be in rescue mode.
+		ctrl.LoggerFrom(ctx).Info("pre-provision: expected rescue system, but found different hostname", "hostname", hostName, "sshOutput", out.String())
+		return actionContinue{delay: 10 * time.Second}
```

Comment:
should we requeue in this case? Is this PR for testing or for merging? For merging, I would expect that we handle this situation in a way that can be done in production. I think requeueing wouldn't be it. Or do you think this might be temporary?

Reply:
I want to merge that PR :-)

I guess this could happen because of a stale cache read by the controller.

If we see that machine hang there in production (which I do not expect), then we should handle that differently.

Reply:
but what is the value of this PR then? I thought it is about handling these cases in production in a better way. Currently, we would make them hang forever (which might have happened anyway of course). 

And which cache would this read from? the hostname on that server?  

Tbh I don't understand the use case for this really.. Can you maybe explain why you wanna do this change?

Reply:
I have seen logs on a customer system which looked like the underlaying system changed during provisioning.

That's why I want to check before connecting to the rescue-system that the rescue system is active.

Reply:
@janiskemper here is a second reason:

failed to reconcile HetznerBareMetalHost ci14972060597-sn76rg/bm-e2e-1724024: action \"ensure-provisioned\" failed: failed to handle incomplete boot - provisioning: unexpected hostname: bm-ci14972069472-kvra9l-control-plane-xh4gj

In our CI we get unexpected hostname because a different CI run uses the same machine. That's why I think that double checking that the rescue system is running makes sense.

Reply:
@janiskemper I updated the PR: 09795339 

If there is a connection error, the code will re-try. If the hostname is unexpected a permanent error gets set.

Reply:
@janiskemper can this thread be resolved?